reframe/core/logging.py

```diff
@@ -689,10 +688,22 @@ def emit(self, record):
             return
 
         try:
-            requests.post(
-                self._url, data=json_record,
-                headers=self._headers
-            )
+            while True:
+                response = requests.post(
+                    self._url, data=json_record,
+                    headers=self._headers
+                )
+                if response.status_code == 200:
+                    break
+
+                if response.status_code == 429:
+                    time.sleep(1)
+                    continue
+
+                raise LoggingError(
+                    f'logging failed: HTTP response code '
+                    f'{response.status_code}'
+                )
```

Comment:
Should we crash in this case, or try to print a warning?

Reply:
How does the outpupt look like when we raise `LoggingError`? I see that currently we do raise it, but I'm not sure it's the right thing. I would rather go for a warning instead (remember to include the name of the handler in the message).